snapshot-labs/snapshot-sidekick
# feat: AI text to speech for proposal body

# AI text to speech 

Generate a .mp3 audio file through OpenAI API from the proposal's body 

NOTE:

- Will work only for body with at most 4096 chars (limit on OpenAI side), and at least 500 chars
- Audio files are cached and saved on AWS after first call

## Test 

- Ensure you have `OPENAI_API_KEY` env var set in your `.env`
- (optional) In `.env`, set `STORAGE_ENGINE` to `file`, to save the mp3 file in `/tmp` locally, for easier access
- Send a curl POST request to: `http://localhost:3005/api/ai/tts/[PROPOSAL-ID]` (choose a proposal with a body length between 500 and 4096 chars)
- It should return a Buffer for an audio file (may take some time if file is not cached yet)
- If you have set your storage engine to file, go to /tmp to find and listen to the audio file
